### ISSUE TITLE
DataSet.update doesn't work for objects from iframe (gwt usage issue)

### ISSUE THREAD

**Issue by AleksejDomash** [almende/vis#2097]
https://github.com/almende/vis/blob/master/lib/DataSet.js checks data parameter in update method using data instanceof Object. It won't work if data has been created in iframe.
Here is explanation:
https://groups.google.com/forum/#!topic/google-web-toolkit/CdgvqajYeDI

Proposal if to move instanceof checks to separate class as util methods and introduce additional check using toString method, like:
Object.prototype.toString.call(o) === '[object Object]

Thanks

**Comment by mojoaxel** [comment]
@AleksejDomash Thanks for reporting this. Could you please provide a simple example (e.g. jsbin) that shows how to reproduce this problem!?

**Comment by ZacBrownBand** [comment]
I ran into this same issue but while using multiple windows. I have a fix and will be opening a pr momentarily. First I wanted to update this issue.

I found a post on SO explaining why this is an issue, 
> Since your window's Object and the source window's Object aren't the same thing, an instance of one won't be an instance of the other. 
[source](http://stackoverflow.com/questions/14391506/instanceof-operator-fails-when-passing-an-object-through-windows)

Here is jsfiddle showing the issue using a iFrame. [Link here](http://jsfiddle.net/5oefapnw/)

In case there is a link break:
```html
<iframe id="other" style="width:500px"></iframe>
```
```js
var other = document.getElementById('other');

window.obj = {};

other.contentWindow.document.write('<pre><script>' +
  'var obj = parent.obj;' +
  'document.write("obj instanceof Object          -> ");' +
  'document.writeln(obj instanceof Object);' +
  'document.write("obj && typeof obj === &quot;object&quot; -> ");' +
  'document.writeln(obj && typeof obj === "object");' +
  '<\/script></pre>');
```

![screen shot 2017-01-25 at 1 45 54 pm](https://cloud.githubusercontent.com/assets/10978390/22304243/a43439a0-e304-11e6-9d9f-08b819ace12b.png)

**Comment by yotamberk** [comment]
closed with #2631. Thanks @ZacBrownBand!
